src/utils.rs

```diff
@@ -19,36 +46,19 @@ pub(crate) fn clear_wallets_vault(path: &Path) -> Result<()> {
     Ok(())
 }
 
-/// Create the `.accounts` file which holds the number of derived accounts so far
-pub(crate) fn create_accounts_file(path: &Path, number_of_derived: usize) -> Result<()> {
-    fs::write(path.join(".accounts"), number_of_derived.to_string())?;
+/// Create the `.accounts` file which holds the addresses of accounts derived so far
+pub(crate) fn create_accounts_file(path: &Path, accounts: Vec<String>) -> Result<()> {
+    let account_file = serde_json::to_string(&Accounts::new(accounts))?;
+    fs::write(path.join(".accounts"), account_file)?;
     Ok(())
 }
 
-/// Read the number of accounts from `.accounts` file
-pub(crate) fn number_of_derived_accounts(path: &Path) -> Result<usize> {
-    let accounts_file_content = fs::read_to_string(path.join(".accounts"));
-    if let Ok(accounts_file_content) = accounts_file_content {
-        Ok(accounts_file_content.parse()?)
+/// Returns the number of the accounts derived so far by reading the .accounts file from given path
+pub(crate) fn number_of_derived_accounts(path: &Path) -> usize {
+    let accounts = Accounts::from_dir(path);
+    if let Ok(accounts) = accounts {
+        accounts.addresses().len()
     } else {
-        Ok(0)
-    }
-}
-
-/// Derives the already created accounts
-pub(crate) fn derived_wallets(path: &Path) -> Result<Vec<Wallet>> {
```

Comment:
We'll want this logic for restoring from a seed phrase. Noting so we don't forget to copy it from here when needed.